node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/containerdata/MotdContainerData.java

```diff
@@ -0,0 +1,95 @@
+                                                 "    local -r no_color='\\e[0m'\n" +
+                                                 "    local -r green='\\e[0;32m'\n" +
+                                                 "## Use red zone name for main prod zones, yellow for other main zones and no colour for cd and dev zones.\n" +
+                                                 "    local -r alert=#if($zone.getSystem() == \"main\")#if($zone.getEnvironment() == \"prod\")'\\e[0;91m'#else'\\e[0;33m'#end#else$green#end\n" +
+                                                 "\n" +
+                                                 "\n" +
+                                                 "    echo -e \"\n" +
+                                                 "${green}Zone          : ${alert}$zone.getSystem().toUpperCase() $zone.getEnvironment().toUpperCase() $zone.getRegion().toUpperCase()\n" +
+                                                 "${green}Node type     : ${no_color}$node.type()\n" +
+                                                 "${green}Host name     : ${no_color}$(hostname)\n" +
+                                                 "${green}Uptime        : ${no_color}$uptime\n" +
+                                                 "${green}Version       : ${no_color}wanted = $node.wanted().orElse(\"unknown\"); installed = $node.installed().orElse(\"unknown\")\n" +
+                                                 "#if($node.owner().isPresent())\n" +
+                                                 "${green}Node state    : ${no_color}$node.state()\n" +
+                                                 "${green}Owner         : ${no_color}$node.owner().get().serializedForm()\n" +
+                                                 "#end\n" +
+                                                 "\"\n" +
+                                                 "}\n" +
+                                                 "\n" +
+                                                 "# Display motd (gently)\n" +
+                                                 "[ ! -f ~/.hushlogin ] && motd\n";
+
+    private final String renderedString;
+
+    public MotdContainerData(ContainerNodeSpec nodeSpec, Environment environment) {
+        renderedString = Template.of(templateString)
+                .set("zone", environment)
+                .set("node", new Node(nodeSpec.nodeType,
+                                      nodeSpec.nodeState,
+                                      nodeSpec.vespaVersion,
+                                      nodeSpec.wantedVespaVersion,
+                                      nodeSpec.owner))
+                .render();
+    }
+
+    public void writeTo(ContainerData containerData) {
+        writeTo(containerData::addFile);
+    }
+
+    void writeTo(BiConsumer<Path, String> fileWriter) {
+        fileWriter.accept(motdPath, renderedString);
+    }
+
+
+    // Needs to be public for Velocity to use it.
+    public static class Node {
```

Comment:
Velocity can't access fields -- only methods. 

It would be less code dropping this class, and just passing the things directly, I suppose ... but then it would be less similar to the MotD we have in _hosted_. 

In hindsight, I think the whole MotD could just as well have been assembled in pure Java, but that's not likeyl to change now :)